lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -314,6 +318,14 @@ def configure(conf)
             @resource.type = DATAFLOW_CONSTANTS[:resource_type]
           elsif @subservice_name == ML_CONSTANTS[:service]
             @resource.type = ML_CONSTANTS[:resource_type]
+          elsif @subservice_name = DATAPROC_CONSTANTS[:service]
```

Comment:
Why do both? If you are automatically detecting Dataproc, you shouldn't need this block.